### DISCUSSION TITLE
On Importing a bank, is it possible to create a BANK at the same time?

### DISCUSSION THREAD

**Discussion by troach242** [christofmuc/KnobKraft-orm#350] [Ideas]
When working with my existing synthesizers, example SQ-80, when I import a bank from a file, an import log is created. This works Ok, but it would be more useful for me if when importing a bank sysex file, a user bank was also created with all of the patches in it. The work was already done by the bank sysex file.. 

A few challenges occur for me:
1. It is nice to have the bank as imported in case I want to dump it all to the synthesizer. without using an external sysex program.
2. Sometimes, duplicate patches exist in the import and it only imports unique patches (awesome), but that is confusing when I want to reuse the "bank".
3. If I want to recreate a user bank from the import, I need to drag each patch one at a time (very time consuming). This is even more difficult when duplicates are skipped in the import and I would need to search/find skipped patches in my library to recreate the bank file that "technically" already exists.

Thank you in advance for your support and time working on this amazing tool.

**Comment by christofmuc** [comment]
That's a great feature idea! It shouldn't be too hard, as banks are stored internally just as lists of pointers to patches in the Library, so effectively it would be just to create a list of type "bank" after the import. It will be harder to figure our which bank was imported - this is not always possible, as some synths just send a stream of edit buffers as bank dump (e.g. Access Virus), and there is no information which bank had been selected. 

But that's certainly easier than having to recreate the bank after import.

On another idea, the Imports feature is still to be migrated to be proper Lists as well, and then you could drag an import list into a newly created bank - that would not be the same as creating the bank right away, but would avoid the drag one by one issue. I need to look and see what is the easiest and most compatible addition!

I will create a proper enhancement issue from this thread.

**Reply by troach242** [reply]
thank you for the quick reply.. I cant wait..

**Reply by christofmuc** [reply]
If we ignore the duplicates problem for a second, I just remembered the new "fill" function for creating user banks or lists.

If you select the import and fill the grid with the imported patches, you can create a new user bank and select the "fill first patches" option to populate the user bank. This will create a user bank putting the patches in the grid into it in the displayed order, no need to drag and drop individually.

Of course, duplicates will be missing in the import list, so this still needs a proper solution.
